### DISCUSSION TITLE
Merge-Knit/Knit-Merge

### DISCUSSION THREAD

**Discussion by nsacco** [quarto-dev/quarto-cli#3577] [Q&A]
Writing a book with Quarto with one file per (*.qmd) per chapter, I am trying to add some references and keys/links by chapter (list of acronyms using a Lua filter). Still, Quarto seems to render each chapter individually. This is similar to bookdown's "Knit-Merge" strategy, as opposed to the "Merge-Knit" where all chapters are combined and generated. Is there any solution to avoid this and define the key once? Also, working with different pages, the links are not recognized, or, does Quarto supports the "Merge-Knit" strategy or something similar?

**Comment by cderv** [comment]
> does Quarto supports the "Merge-Knit" strategy or something similar?

Regarding this topic of KM vs MK approach in **bookdown** , discussion happens already in https://github.com/quarto-dev/quarto-cli/discussions/1045

Quarto does not have such approach for now. Design is different. 
To have a single document before rendering, you would need to leverage maybe [include](https://quarto.org/docs/authoring/includes.html#overview) shortcodes to create one single document before rendering. But this would not be a book format at the end I think (several files vs single file)

> Quarto seems to render each chapter individually.

Yes - Quarto does not merge input before rendering with Pandoc, and each file is rendered individually. 

> I am trying to add some references and keys/links by chapter (list of acronyms using a Lua filter)

Our cross reference system is working already across chapters. Are you trying to customise it ? What are you trying to do exactly ? 
Knowing in more details would allow us to think better on how to improve our Lua internal system to allow such approach. 

> Also, working with different pages, the links are not recognized,

Links should works across chapters following [cross references](https://quarto.org/docs/authoring/cross-references.html) mechanism.
Can you develop with an example what is failing ? (you can open a new issue or new question thread for this)
